cfme/rest/gen_data.py

```diff
@@ -160,10 +163,17 @@ def _order_finished():
         return service_request.request_state.lower() == 'finished'
 
     wait_for(_order_finished, num_sec=2000, delay=10)
-    assert 'error' not in service_request.message.lower(), \
-        'Provisioning failed with the message `{}`'.format(service_request.message)
-
-    service_name = str(service_request.options['dialog']['dialog_service_name'])
+    assert 'error' not in service_request.message.lower(), ('Provisioning failed: `{}`'
+                                                            .format(service_request.message))
+
+    service_name = VersionPicker({
+        Version.lowest(): lambda: re.search(r'\[({}[0-9-]*)\] '
```

Comment:
Sorry to harp on the "centralized helper method" thing, since 2 other reviewers have too, but this is basically the same as `def _get_dialog_service_name` in test_rest_services right? So should that helper live in utils somewhere?